lib/app/features/auth/views/pages/two_fa/twofa_options_step.dart

```diff
@@ -73,19 +74,9 @@ class TwoFAOptionsStep extends HookConsumerWidget {
                       ),
                       SizedBox(height: 16.0.s),
                       Button(
-                        onPressed: () {
-                          final optionsAmount = optionsState.optionsAmount;
-                          final hasValues =
-                              optionsState.selectedValues.nonNulls.length == optionsAmount;
-                          final isValidValues =
-                              !optionsState.selectedValues.contains(TwoFaType.auth) ||
-                                  optionsState.selectedValues.nonNulls.length > 1;
-
-                          if ((hasValues && isValidValues) ||
-                              formKey.value.currentState!.validate()) {
-                            onConfirm();
-                          }
-                        },
+                        disabled: !isFormValid,
+                        type: isFormValid ? ButtonType.primary : ButtonType.disabled,
+                        onPressed: !isFormValid ? null : onConfirm,
```

Comment:
Just avoid negative condition:
```suggestion
                        onPressed: isFormValid ? onConfirm : null,
```